jwst/ramp_fitting/ramp_fit.py

```diff
@@ -105,9 +118,286 @@ def ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
+            pixeldq_slice = input_model.pixeldq[ i * rows_per_slice: (i + 1) * rows_per_slice, :].copy()
+
+            slices.insert(i, (data_slice, err_slice, groupdq_slice, pixeldq_slice, buffsize, save_opt, readnoise_slice,
+                            gain_slice, weighting,
+                            input_model.meta.instrument.name, input_model.meta.exposure.frame_time,
+                            input_model.meta.exposure.ngroups, input_model.meta.exposure.group_time,
+                            input_model.meta.exposure.groupgap, input_model.meta.exposure.nframes,
+                            input_model.meta.exposure.drop_frames1, int_times))
+
+        # last slice gets the rest
+        readnoise_slice = readnoise_2d[(number_slices - 1) * rows_per_slice: total_rows, :]
+        gain_slice = gain_2d[(number_slices - 1) * rows_per_slice: total_rows, :]
+        data_slice = input_model.data[:, :, (number_slices - 1) * rows_per_slice: total_rows, :].copy()
+        err_slice = input_model.err[:, :, (number_slices - 1) * rows_per_slice: total_rows, :].copy()
+        groupdq_slice = input_model.groupdq[:, :, (number_slices - 1) * rows_per_slice: total_rows, :].copy()
+        pixeldq_slice = input_model.pixeldq[(number_slices - 1) * rows_per_slice: total_rows, :].copy()
+        slices.insert(number_slices - 1, (data_slice, err_slice, groupdq_slice, pixeldq_slice, buffsize, save_opt,
+                                          readnoise_slice, gain_slice, weighting,
+                                          input_model.meta.instrument.name, input_model.meta.exposure.frame_time,
+                                          input_model.meta.exposure.ngroups, input_model.meta.exposure.group_time,
+                                          input_model.meta.exposure.groupgap, input_model.meta.exposure.nframes,
+                                          input_model.meta.exposure.drop_frames1, int_times))
+
+        log.debug("Creating %d processes for ramp fitting " % number_slices)
+        #start up the processes for each slice
+        real_results = pool.starmap(ols_ramp_fit, slices)
+        pool.close()
+        pool.join()
+        k = 0
+        log.debug("All processes complete")
+        # Create new model for the primary output.
+        actual_segments = real_results[0][20]
+        actual_CRs = real_results[0][21]
+        int_model, opt_model, out_model = create_output_models(input_model,
+                            number_of_integrations, save_opt, total_cols, total_rows,
+                                                               actual_segments, actual_CRs)
+        # iterate over the number of slices and place the results into the output models
+        for resultslice in real_results:
+            if len(real_results) == k + 1:  # last result
+                out_model.data[k * rows_per_slice: total_rows, :] = resultslice[0]
```

Comment:
Perhaps mapping the index to resultslice something more descriptive would help clarify this.

Reply:
I have changed all the indices to use explicit variables to make it easier to read.